experiments/sample_ssl4eo.py

```diff
@@ -0,0 +1,265 @@
+        minx, miny, maxx, maxy = self.boundaries.bounds
+        lon = np.random.uniform(minx, maxx)
+        lat = np.random.uniform(miny, maxy)
+        p = Point(lon, lat)
+        if self.boundaries.contains(p):
+            return [p.x, p.y]
+        else:
+            return self.sample_point()
+
+    @staticmethod
+    def get_country_boundaries(
+        download_root: str = os.path.expanduser("~/.cache/naturalearth"),
+    ) -> shape:
+        url = "https://www.naturalearthdata.com/http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_countries.zip"  # noqa: E501
+        filename = "ne_110m_admin_0_countries.shp"
+        if not os.path.exists(os.path.join(download_root, os.path.basename(url))):
+            download_and_extract_archive(url, download_root)
+        sf = shapefile.Reader(os.path.join(download_root, filename))
+        return shape(sf.shapes().__geo_interface__)
+
+
+class OverlapError(Exception):
+    pass
+
+
+def sample_random_locations_rtree(
+    idx: int,
+    sampler: Any,
+    radius: float,
+    overlap_ratio: float,
+    rtree_obj: index.Index = index.Index(),
+) -> List[float]:
+    """sample new coord and check overlap --- rtree"""
+    # use rtree to avoid strong overlap
+    count = 0
+    bbox_radius = (1 - overlap_ratio) * radius / 1000
+    while count < 1:
+        new_coord = sampler.sample_point()  # (lon,lat) of top-10000 cities
+        bbox = (
+            new_coord[0] - sampler.km2deg(bbox_radius),
+            new_coord[1] - sampler.km2deg(bbox_radius),
+            new_coord[0] + sampler.km2deg(bbox_radius),
+            new_coord[1] + sampler.km2deg(bbox_radius),
+        )
+        if list(rtree_obj.intersection(bbox)):
+            continue
+        rtree_obj.insert(idx, bbox)
+        count += 1
+        center_coord = [new_coord[0], new_coord[1]]
+    return center_coord
+
+
+def fix_random_seeds(seed: int = 42) -> None:
+    np.random.seed(seed)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--save_path", type=str, default="./data/", help="dir to save data"
+    )
+    parser.add_argument(
+        "--radius", type=float, default=1320, help="patch radius in meters"
+    )
+    parser.add_argument(
+        "--overlap_ratio",
+        type=float,
+        default=0.25,
+        help="max overlap ratio between adjacent patches",
+    )
+    parser.add_argument(
+        "--num_cities", type=int, default=10000, help="number of cities to sample"
+    )
+    parser.add_argument(
+        "--std", type=int, default=50, help="std of gaussian distribution"
+    )
+    parser.add_argument(
+        "--resume", type=str, default=None, help="resume from a previous run"
+    )
+    parser.add_argument(
+        "--indices_range",
+        type=int,
+        nargs=2,
+        default=[0, 250000],
+        help="indices to download",
+    )
+
+    args = parser.parse_args()
+
+    fix_random_seeds(seed=42)
```

Comment:
```suggestion
    np.random.seed(42)
```
Now you can delete the function and save 4 lines of code.